test-toolkit/src/main/java/org/bonitasoft/test/toolkit/bpm/TestHumanTask.java

```diff
@@ -208,11 +202,19 @@ private TestHumanTask waitState(final APISession apiSession, final String state)
         return this;
     }
 
-    public TestHumanTask waitState(final TestUser initiator, final String state) {
-        return waitState(initiator.getSession(), state);
-    }
-
-    public TestHumanTask waitState(final String state) {
-        return waitState(TestToolkitCtx.getInstance().getInitiator(), state);
+    public TestHumanTask waitFinished() {
```

Comment:
It was not working because it was only searching for task in state in non archived elements